float8_experimental/float8_dynamic_utils.py

```diff
@@ -130,20 +146,38 @@ def unwrap(t):
         )
 
     def __tensor_flatten__(self):
-        return ["_tensor"], self._mm_config
+        if self._precomputed_amax:
+            return ["_tensor", "_precomputed_amax"], self._mm_config
```

Comment:
does having `Optional[torch.Tensor]` as a subclass field work with torch.compile? Or do we not care about torch.compile in this code path?

Reply:
torch.compile assumes every tensor from `__tensor_flatten__` is not None. That's why I added `if-else` so when 
`_precomputed_amax is None`, it won't be included in `__tensor_flatten__`

with the `if-else`, torch.compile works. I verified it here and reported perf gaps with eager https://github.com/pytorch/pytorch/issues/129457